projects/hmcts-annotation-ui-lib/src/lib/viewers/annotation-pdf-viewer/comments/comment-item/comment-item.component.spec.ts

```diff
@@ -0,0 +1,298 @@
+      expect(component.annotationHeight).toBe(rectangleTop.height);
+      expect(component.annotationLeftPos).toBe(rectangleBottom.x);
+    });
+
+    it('should emit commentRendered event emitter on dataloaded', (done) => {
+      component.commentRendered.subscribe((commentRendered: boolean) => {
+        expect(commentRendered).toBeTruthy();
+        done();
+     });
+      component.ngOnInit();
+    });
+  });
+
+  describe('onDestroy', () => {
+    it('should unsubscribe from the commentBtnSub', () => {
+      const commentBtnSub = component['commentBtnSub'];
+      spyOn(commentBtnSub, 'unsubscribe');
+      component.ngOnDestroy();
+      expect(commentBtnSub.unsubscribe).toHaveBeenCalled();
+    });
+  });
+
+  describe('handleShowBtn', () => {
+    it('should set hideButton to false', () => {
+      component.handleShowBtn();
+      expect(component['hideButton']).toBeFalsy();
+    });
+  });
+
+  describe('handleHideBtn', () => {
+    it('should set hideButton to true', () => {
+      component.handleHideBtn();
+      expect(component['hideButton']).toBeTruthy();
+    });
+  });
+
+  describe('handleCommentClick', () => {
+    it('should update the commentBtn subject with its own comment ID', () => {
+      spyOn(mockAnnotationStoreService, 'setCommentBtnSubject');
+      spyOn(mockAnnotationStoreService, 'setAnnotationFocusSubject');
+
+      const mockEvent = {
+        stopPropagation() {}
+      };
+      component.handleCommentClick(mockEvent);
+
+      expect(mockAnnotationStoreService.setCommentBtnSubject)
+        .toHaveBeenCalledWith(component.comment.id);
+    });
+  });
+
+  describe('convertFormToComment', () => {
+    it('should convert form to comment', async(() => {
+      const actual = component.convertFormToComment(commentForm);
+      expect(actual instanceof Comment).toBeTruthy();
+      expect(actual.id).toEqual(comment.id);
+      expect(actual.annotationId).toEqual(comment.annotationId);
+      expect(actual.content).toEqual(commentForm.value.content);
+    }));
+  });
+
+  describe('onSubmit', () => {
+    it('should add current date to comment onSubmit', async(() => {
+      component.commentItem = commentForm;
+      component.onSubmit();
+      expect(mockAnnotationStoreService.comment.createdDate).toBeTruthy();
+    }));
+  });
+
+  describe('onBlur', () => {
+    it('should run on blur', async(() => {
+      component.onBlur();
+    }));
+  });
+
+  describe('handleDeleteComment', () => {
+    it('should call delete comment when handleDeleteComment is called', async(() => {
+      spyOn(mockAnnotationStoreService, 'deleteComment');
+      component.handleDeleteComment();
+      expect(mockAnnotationStoreService.deleteComment).toHaveBeenCalled();
+    }));
+  });
+
+  describe('isModified', () => {
+    it('should return false if createdDate is null', async(() => {
+      mockComment.createdDate = null;
+      component.comment = mockComment;
+      const modified = component.isModified();
```

Comment:
this is being repeated, maybe best to have it outside the it and call it from there in the other test.